airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -171,6 +186,7 @@ def _is_relevant_upstream(upstream: TaskInstance) -> bool:
             for finished_ti in dep_context.ensure_finished_tis(ti.get_dagrun(session), session)
             if _is_relevant_upstream(finished_ti)
         )
+        finished_upstream_tis = list(finished_upstream_tis)
```

Comment:
Why not do a list comprehension right above here instead? Does this actually need to be a list?

Reply:
I'm pretty sure this isn't necessary, I've removed it.